src/lndk_offers.rs

```diff
@@ -98,6 +167,42 @@ pub(crate) async fn create_request_invoice(
     .unwrap()
 }
 
+// send_invoice_request sends the invoice request to node that created the offer we want to pay.
+pub async fn send_invoice_request(
+    network: Network,
+    invoice_request: InvoiceRequest,
+    onion_messenger: impl OnionMessageHandler + SendOnion,
+    intermediate_nodes: Vec<PublicKey>,
+    blinded_path: BlindedPath,
+) -> Result<(), OfferError<bitcoin::secp256k1::Error>> {
+    // We need to make sure our local onion messenger is connected to the needed local peers in
+    // so it can relay along the onion messages to them.
+    let onion_message_optional: u64 = 1 << ONION_MESSAGES_OPTIONAL;
+    let init_features = InitFeatures::from_le_bytes(onion_message_optional.to_le_bytes().to_vec());
+    let network = vec![ChainHash::using_genesis_block(network)];
+    onion_messenger
+        .peer_connected(
+            &intermediate_nodes[0],
+            &Init {
+                features: init_features,
```

Comment:
Two high level questions here. 

One: So it seems that send_onion_message requires that we supply a full path to the offer creator, including a path of intermediate nodes *to* the blinded path as well as the blinded path. Here we're connecting directly to the introduction node so we can just send the onion message directly. But idk, maybe if the introduction node isn't accepting direct peer connects (?) or something, we could find a route to the introduction node?

Two: the peer we're connecting to here is the introduction node provided in the offer, so we can assume it provides onion messaging support correct? Or should we double check that it advertises support for good measure?

Reply:
Cool, leaving that pathfinding step as a follow up sounds good - I'll open an issue as a reminder. 

> Makes sense to me to have a validation step where we sanity check this, because there's no point proceeding if they couldn't make a good path for us.

Good point, added a check for that.

Ok so I made some changes. It might seem a little weird that we use another onion messenger instance here... it's kind of unnecessary, but one key thing it provides is turning our InvoiceRequest into a proper onion message that we can send. It looks like LDK will export the `create_onion_message` fn in the next release so we can use it directly... So when that's out, we could get rid of the onion messenger instance here and just directly use lnd's SendCustomMessage call instead, but using the onion messenger here is the only workaround I could think of for now